[KOGITO-8517] - Support external resources configuration for WorkflowCR in devmode (#81)
kiegroup/kogito-serverless-operator: README.md

make undeploy
```

## Use Workflow Development Mode

> In the development mode, a user can edit and reload the files using the quarkus dev mode.

Reviewer: need to sync this with the work that @davidesalerno is doing about exposing the dev mode switch in the CR.

Author: There aren't conflicts

make undeploy
```

## Use Workflow Development Mode
In the development mode, a user can edit and reload the files using the quarkus dev mode.

> To enable the dev mode add this annotation in the KogitoServerlessWorkflow CR:

Reviewer 2: ## Use Workflow Development Mode
In the development mode, a user can edit and reload the files using the quarkus dev mode.

To enable the dev mode add this annotation in the KogitoServerlessWorkflow CR:

```
sw.kogito.kie.org/profile: dev
```
in the kogitoserverlessworkflow CR

> The external files that can be edited during the dev mode
> must be saved in a configmap per type, currently CamelRoute, OpenAPI and Async API are supported,
> to reference the configmaps in the labels section must be added using the following keys and your preferred name as values,
> the same label must be added on the configmap.
> Currently the SWF app handle Camel, OpenAPI and AsyncAPI, to handle other type is up to the user add the dependencies to handle other types
> 

Reviewer 2: ```suggestion
In development mode, different external files can be edited. Each file type will be saved into a different configmap.  Currently we support those types:
- CamelRoute
- OpenAPI 
- Async API
- Generic: any other type

To be able to edit a file via a configmap, you must setup a label which will reference the specific configmap:
```

```

> labels:
> .. 
>   sw.kogito.kie.org/resource-camel: mycamel-configmap
>   sw.kogito.kie.org/resource-openapi: myopenapi-configmap
>   sw.kogito.kie.org/resource-asyncapi: myasyncapi-configmap
>   sw.kogito.kie.org/resource-generic: mygeneric-configmap

Reviewer 2: ```suggestion
  sw.kogito.kie.org/resource-camel: mycamel-configmap
  sw.kogito.kie.org/resource-openapi: myopenapi-configmap
  sw.kogito.kie.org/resource-asyncapi: myasyncapi-configmap
  sw.kogito.kie.org/resource-generic: mygeneric-configmap
```

> ## Use Workflow Development Mode
> In the devmode a user can edit and reload the files using the quarkus devmode
> To enable the dev mode add the annotation
> ```
> sw.kogito.kie.org/profile: dev
> ```
> in the kogitoserverlessworkflow CR
> 
> The external files that can be edited during the dev mode
> must be saved in a configmap per type, currently CamelRoute, OpenAPI and Async API are supported,
> to reference the configmaps in the labels section must be added using the following keys and your preferred name as values,
> the same label must be added on the configmap.
> Currently the SWF app handle Camel, OpenAPI and AsyncAPI, to handle other type is up to the user add the dependencies to handle other types
> 
> ```
> labels:
> .. 
>   sw.kogito.kie.org/resource-camel: mycamel-configmap
>   sw.kogito.kie.org/resource-openapi: myopenapi-configmap
>   sw.kogito.kie.org/resource-asyncapi: myasyncapi-configmap
>   sw.kogito.kie.org/resource-generic: mygeneric-configmap
> ```
> Every entry in these configmaps will be turned into a file inside /src/main/resources/<type specific folder> and using the key as file name
> 

Reviewer 2: ```suggestion
## Use Workflow Development Mode
In the development mode, a user can edit and reload the files using the quarkus dev mode.

To enable the dev mode add this annotation in the KogitoServerlessWorkflow CR:
```
  sw.kogito.kie.org/profile: dev
```

In development mode, different external files can be edited. Each file type will be saved into a different configmap.  Currently we support those types:
- CamelRoute
- OpenAPI 
- Async API
- Generic: any other type

To be able to edit a file via a configmap, you must setup a label which will reference the specific configmap:

```
  sw.kogito.kie.org/resource-camel: mycamel-configmap
  sw.kogito.kie.org/resource-openapi: myopenapi-configmap
  sw.kogito.kie.org/resource-asyncapi: myasyncapi-configmap
  sw.kogito.kie.org/resource-generic: mygeneric-configmap
```

Each label entry will be turned into a file inside the path `src/main/resources/<type specific folder>` and using the key as the filename.
```

Reviewer 2: and GH is failing with the review comment ... Please make sure you take the overall ...

> Every entry in these configmaps will be turned into a file inside /src/main/resources/<type specific folder> and using the key as file name

Reviewer 2: ```suggestion
Each label entry will be turned into a file inside the path `src/main/resources/<type specific folder>` and using the key as the filename.
```

make undeploy
```

## Use Workflow Development Mode
In the development mode, a user can edit and reload the files using the quarkus dev mode.
To enable the dev mode add this annotation in the KogitoServerlessWorkflow CR:
```
sw.kogito.kie.org/profile: dev
```

In development mode, different external files can be edited. Each file type will be saved into a different configmap.  Currently we support those types:
- CamelRoute
- OpenAPI
- Async API
- Generic: any other type

>   To be able to edit a file via a configmap, you must setup an annotation which will reference the specific configmap previously created on the cluster.

Author: ok

```
  sw.kogito.kie.org/resource-camel: mycamel-configmap
  sw.kogito.kie.org/resource-openapi: myopenapi-configmap
  sw.kogito.kie.org/resource-asyncapi: myasyncapi-configmap
  sw.kogito.kie.org/resource-generic: mygeneric-configmap
```
Each annotation entry will be turned into a file inside the path `src/main/resources/<type specific folder>` and using the key as the filename.

You could see an example of configmap and a workflow with one of this annotation in the file
```

> config/samples/sw.kogito_v1alpha08_kogitoserverlessworkflow_devmodeWithConfigMapAndExternalResource.yaml:

Author: ok

make undeploy
```

> ## Use Workflow Development Mode

Reviewer 2: please update the TOC as well

> In the development mode, a user can edit and reload the files using the quarkus dev mode.
> To enable the dev mode add this annotation in the KogitoServerlessWorkflow CR:

Reviewer 2: ```suggestion
In the development mode, a user can edit and reload the files using the quarkus dev mode.  
To enable the dev mode add this annotation into the KogitoServerlessWorkflow CR:
```

```
sw.kogito.kie.org/profile: dev
```

> 

Reviewer 2: ```suggestion
### Development mode: External files

```

> In development mode, different external files can be edited. Each file type will be saved into a different configmap.  

Reviewer 2: ```suggestion
In development mode, different external files can be edited.    
```

Author: ok

> Currently we support those types:

Reviewer 2: ```suggestion
Currently we support those file types:
```

Author: ok

- CamelRoute
- OpenAPI
- Async API
- Generic: any other type

> To be able to edit a file via a configmap, you must setup an annotation which will reference the specific configmap previously created on the cluster.

Reviewer 2: ```suggestion
Each file type will be saved into a different configmap.  
To be able to edit a file via a configmap, you must setup an annotation which will reference the specific configmap previously created on the cluster.
```

```
  sw.kogito.kie.org/resource-camel: mycamel-configmap
  sw.kogito.kie.org/resource-openapi: myopenapi-configmap
  sw.kogito.kie.org/resource-asyncapi: myasyncapi-configmap
  sw.kogito.kie.org/resource-generic: mygeneric-configmap
```

> Each annotation entry will be turned into a file inside the path `src/main/resources/<type specific folder>` and using the key as the filename.

Reviewer 2: doc matter: Where can I find the `type specific folder` ?

Author: Added

make undeploy
```

## Use Workflow Development Mode
In the development mode, a user can edit and reload the files using the quarkus dev mode. 
To enable the dev mode add this annotation into the KogitoServerlessWorkflow CR:
```
sw.kogito.kie.org/profile: dev
```

### Development mode External files

> In development mode, different external files can be edited.
> Currently we support those file types:
> - CamelRoute
> - OpenAPI
> - Async API
> - Generic: any other type
> 
> Each file type will be saved into a different configmap.  
> To be able to edit a file via a configmap, you must setup an annotation which will reference the specific configmap previously created on the cluster.
> 
> Each annotation entry will be turned into a file inside the path `src/main/resources/<File type specific folder>` and using the key as the filename.

Reviewer 2: ```suggestion
In development mode, different external files can be edited.  
Each file type should be saved into a different configmap.  
In order to be able to edit a file via a configmap, you must set up an annotation in your KogitoServerlessWorkflow CR which reference the specific configmap which is already existing into the namespace.  
Then, each annotation entry will be turned into a file inside the path `src/main/resources/<File type specific folder>` and using the key as the filename.

See the table below for the supported types and the corresponding annotations and folder paths.
```

File type | Annotation | Folder path
---------|-------------------------------------|---

> Camel    | sw.kogito.kie.org/resource-camel    | src/main/resources/routes

Author: yep
